modules/mongodb/mongodb.go

```diff
@@ -0,0 +1,114 @@
+package mongo
+
+import (
+	"context"
+	"time"
+
+	"github.com/netdata/go.d.plugin/agent/module"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func init() {
+	module.Register("mongodb", module.Creator{
+		Defaults: module.Defaults{
+			UpdateEvery:        module.UpdateEvery,
+			AutoDetectionRetry: module.AutoDetectionRetry,
+			Priority:           module.Priority,
+			Disabled:           false,
+		},
+		Create: func() module.Module { return New() },
+	})
+}
+
+func New() *Mongo {
+	return &Mongo{
+		Config: Config{
+			Timeout: defaultTimeout,
+			Uri:     defaultUri,
+		},
+		charts: &module.Charts{},
+	}
+}
+
+type Mongo struct {
+	module.Base
+	Config `yaml:",inline"`
+	client *mongo.Client
+	charts *module.Charts
+}
+
+func (m *Mongo) Init() bool {
+	m.Infof("initializing mongodb")
+	if m.Uri == "" {
+		m.Errorf("connection URI is empty")
+		return false
+	}
+
+	var err error
+	m.charts, err = m.initCharts()
+	if err != nil {
+		m.Errorf("init charts: %v", err)
+		return false
+	}
+	return true
+}
+
+func (m *Mongo) Check() bool {
+	return len(m.Collect()) > 0
+}
+
+func (m *Mongo) Charts() *module.Charts {
+	return m.charts
+}
+
+func (m *Mongo) Collect() map[string]int64 {
+	if m.client == nil {
+		var err error
+		err = m.initMongoClient()
+		if err != nil {
```

Comment:
```suggestion
		if err := m.initMongoClient(); err != nil {
```

How about simplifying it?  We don't use `err` later in the code.

Reply:
done